File: backend/DataAccess/Entities/Expense.cs
using DataAccess.Enums;

namespace DataAccess.Entities
{
    public class Expense
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public ExpenseCategory Category { get; set; }
        public decimal Amount { get; set; }
        public DateTime Date { get; set; }
        public string? Description { get; set; }
        public int TripId { get; set; }

        // Navigation
        public Trip Trip { get; set; } = null!;
    }
}

File: frontend/src/models/Expense.ts
import { ExpenseCategory } from './enums';

export interface Expense {
  id: number;
  name: string;
  category: ExpenseCategory;
  amount: number;
  date: string;
  description?: string;
  tripId: number;
}

export interface CreateExpenseRequest {
  name: string;
  category: ExpenseCategory;
  amount: number;
  date: string;
  description?: string;
}

export interface UpdateExpenseRequest {
  name: string;
  category: ExpenseCategory;
  amount: number;
  date: string;
  description?: string;
}